File: chapt2/chapt2_ws/src/demo_python_pkg/demo_python_pkg/person_node.py
import rclpy
from rclpy.node import Node

class PersonNode(Node): 
    def __init__(self,
                 node_name:str,
                 name: str,
                 age: int) -> None:
        print('PersonNode __init__ 方法被调用了，添加了两个属性')
        super().__init__(node_name)
        self.name = name
        self.age = age
    
    def eat(self, food_name: str):
        """
        eat 的 Docstring
        
        :param self: 说明
        :param food_name: 说明
        :type food_name: str
        """
        # print(f"{self.name}, {self.age}, 爱吃{food_name}")
        self.get_logger().info(f"{self.name}, {self.age}, 爱吃{food_name}")

def main():
    rclpy.init()
    node = PersonNode('zhangsan', '法外狂徒张三', 18)
    node1 = PersonNode('lisi', '法外狂徒李四', 89)
    node.eat('yuxiangrousi')
    node1.eat('西红柿')
    rclpy.spin(node)
    rclpy.spin(node1)
    rclpy.shutdown()

File: chapt2/chapt2_ws/src/demo_cpp_pkg/src/person_node.cpp
#include "rclcpp/rclcpp.hpp"

class PersonNode : public rclcpp::Node
{
private:
    // 声明
    std::string name_;
    int age_;

public:
    PersonNode(const std::string &node_name, const std::string &name, const int &age)
        : Node(node_name) /*调用父类的构造函数*/
    {
        // 赋值
        this->name_ = name;
        this->age_ = age;
    }

    void eat(const std::string &food_name)
    {
        RCLCPP_INFO(this->get_logger(), "我是%s,%d, 爱吃%s", this->name_.c_str(), this->age_, food_name.c_str());
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<PersonNode>("PersonNode", "李斯", 18);
    RCLCPP_INFO(node->get_logger(), "ni hao c++ node");
    node->eat("yuxiangrousi");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}